redox/redox.go

```diff
@@ -306,116 +321,76 @@ func (h *Handler) RescheduleSubscriptionOrdersForPatient(ctx context.Context, pa
-		match := result.Patients[0]
-		if err := h.patients.UpdateEHRSubscription(ctx, match.ClinicId.Hex(), *match.UserId, *update); err != nil {
+	// Update the subscription for matched patient only if single match was found
+	if matchOrder.SubscriptionUpdate != nil && len(matchingPatients) == 1 {
+		match := matchingPatients[0]
+		err = h.patients.UpdateEHRSubscription(ctx, clinic.Id.Hex(), *match.UserId, *matchOrder.SubscriptionUpdate)
+		if err != nil {
 			return nil, err
 		}
 	}
 
-	return result.Patients, err
+	return &MatchResult{
+		Clinic:   *clinic,
+		Patients: matchingPatients,
+	}, nil
 }
 
-func (h *Handler) FindMatchingPatientsForAccountCreationOrder(ctx context.Context, clinic clinics.Clinic, order models.NewOrder) ([]*patients.Patient, error) {
-	criteria, err := GetPatientMatchingCriteriaFromNewOrder(order, clinic)
+func (h *Handler) findMatchingPatients(ctx context.Context, clinic clinics.Clinic, matchOrder MatchOrder) ([]*patients.Patient, error) {
+	criteria, err := GetPatientMatchingValuesFromNewOrder(matchOrder.Order, clinic)
 	if err != nil {
 		return nil, err
 	}
 	if criteria == nil {
 		return nil, nil
 	}
 
+	unique := map[string]struct{}{}
 	var matchingPatients []*patients.Patient
 
-	clinicId := clinic.Id.Hex()
-	page := store.Pagination{
-		Offset: 0,
-		Limit:  100,
-	}
-
-	filter := patients.Filter{
-		ClinicId: &clinicId,
-		Mrn:      &criteria.Mrn,
-	}
-	result, err := h.patients.List(ctx, &filter, page, nil)
+	filters, err := criteria.GetFilters(clinic.Id.Hex(), matchOrder.PatientAttributes)
 	if err != nil {
 		return nil, err
 	}
+	for _, filter := range filters {
+		// When matches are non-unique, the results are not useful to clients.
+		// It's unlikely that the count of matched patients will be more than a few,
+		// so querying for 100 will make it obvious that something is wrong
+		page := store.Pagination{
+			Offset: 0,
+			Limit:  100,
+		}
 
-	unique := map[string]struct{}{}
-	if result.TotalCount > 0 {
-		for _, patient := range result.Patients {
-			if patient == nil || patient.UserId == nil {
-				continue
-			}
-			if _, ok := unique[*patient.UserId]; ok {
-				continue
-			}
-			unique[*patient.UserId] = struct{}{}
-			matchingPatients = append(matchingPatients, patient)
+		result, err := h.patients.List(ctx, &filter, page, nil)
+		if err != nil {
+			return nil, err
 		}
-	}
 
-	filter = patients.Filter{
-		ClinicId:  &clinicId,
-		BirthDate: &criteria.DateOfBirth,
-		FullName:  &criteria.FullName,
-	}
-	result, err = h.patients.List(ctx, &filter, page, nil)
-	if err != nil {
-		return nil, err
-	}
-	if result.TotalCount > 0 {
-		for _, patient := range result.Patients {
-			if patient == nil || patient.UserId == nil {
-				continue
+		if result.TotalCount > 0 {
+			for _, patient := range result.Patients {
+				if patient == nil || patient.UserId == nil {
+					continue
+				}
+				if _, ok := unique[*patient.UserId]; ok {
```

Comment:
```suggestion
				if _, found := unique[*patient.UserId]; found {
```
`ok` is used in a lot of examples like this, and it's fine, but I've always felt that tailoring the term more specifically to the situation can help the reader better understand the intention.